Improve config process and error message

At present, if a developer fails to provide a config file for pdp, you get this obnoxious error message:

```python
../../../virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pdp/__init__.py:77: in wrap_auth

    config = get_config()

../../../virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pdp/__init__.py:28: in get_config

    with open(config_filename) as f:

E   IOError: [Errno 2] No such file or directory: '/var/www/dataportal/config.yaml'
```

That stops you in your tracks. I think that we should consider either providing an in-package resource file for the default config, and just give a warning if it falls back to using it. Or at the very least, we should catch this `IOError` and give the user instructions on the fact that they need to provide a config file.

--- a/pdp/__init__.py
+++ b/pdp/__init__.py
@@ -5,7 +5,7 @@
 
 __all__ = ['get_config', 'wrap_auth']
 
-import os
+import sys, os
 from os.path import dirname
 import atexit
 
@@ -25,8 +25,13 @@ from pdp.portals import updateConfig
 
 def get_config():
     config_filename = os.environ.get('PDP_CONFIG', '/var/www/dataportal/config.yaml')
-    with open(config_filename) as f:
-        config = yaml.load(f)
+    try:
+        with open(config_filename) as f:
+            config = yaml.load(f)
+    except IOError:
+        print("pdp/__init__.py: An error occurred while trying to read the config file. Please make sure that the PDP_CONFIG env variable has been set and points to a valid file.")
+        sys.exit(1)
+
     global_config = {
         'css_files': [
             'css/jquery-ui-1.10.2.custom.css',
@@ -68,7 +73,6 @@ def clean_session_dir(session_dir, should_I):
     if should_I and os.path.exists(session_dir):
         print('Removing session directory {}'.format(session_dir))
         rmtree(session_dir)
-
 
 # auth wrappers
 def wrap_auth(app, required=False):

